src/server/pkg/obj/cache_client.go

```diff
@@ -84,19 +108,6 @@ func (c *cacheClient) IsRetryable(err error) bool {
 	return c.fast.IsRetryable(err) || c.slow.IsRetryable(err)
 }
 
-func (c *cacheClient) deleteFromCache(ctx context.Context, p string) error {
```

Comment:
@Tryneus Stale reads are technically possible (even with this, a different client could perform the deletion), but if we ever get into a situation where that happens then something else went wrong in Pachyderm. I am leaning on the deletion of the commit metadata to prevent this from happening and on the eviction to eventually remove it from the cache.